x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/RestGetOperationModeAction.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.indexlifecycle.action;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.core.indexlifecycle.action.GetMaintenanceModeAction;
+import org.elasticsearch.xpack.indexlifecycle.IndexLifecycle;
+
+public class RestGetOperationModeAction extends BaseRestHandler {
+
+    public RestGetOperationModeAction(Settings settings, RestController controller) {
+        super(settings);
+        controller.registerHandler(RestRequest.Method.GET,
+            IndexLifecycle.BASE_PATH + "maintenance", this);
```

Comment:
I think the endpoint shouldn't contain `maintenance` otherwise its confusing what it means when this API returns `mode: normal`. Instead I think this should be `_xapck/index_lifecycle/mode` or `_xapck/index_lifecycle/operation_mode`